### PR TITLE
package.json: Add flow check commands for iOS and Android

### PR DESCRIPTION
**Pull request by empyrical**
This PR adds these two scripts to `package.json`:

 * `flow-check-ios` - for running `flow check` on `.ios.js` files
 * `flow-check-android` - for running `flow check` on `.android.js` files

The Android command makes use of the new `--flowconfig-name` option added to Flow in `0.80.0`

Test Plan:
----------
Both commands have been ran and work as expected.

Release Notes:
--------------
[GENERAL] [ENHANCEMENT] [package.json] - Added flow check commands for iOS and Android

### CODE REVIEW

**Comment by empyrical** [comment]
Added!

File: .circleci/config.yml [hunk 1]
```diff
@@ -187,7 +187,8 @@ aliases:
 
   - &run-flow-checks
     name: Check for errors in code using Flow
-    command: yarn flow check
+    command: yarn flow-check-ios
+    command: yarn flow-check-android
```
Comment (TheSavior) [inline .circleci/config.yml]: Woah! We weren't checking this!? Scary!

Comment (TheSavior) [inline .circleci/config.yml]: I think the second command overwrites the first. Looking at the CI logs for this step in this PR show that it only ran flow on android: https://circleci.com/gh/facebook/react-native/53561?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

You'll probably need to change this from `run-flow-checks` to `run-flow-ios` and have another one for android.

Comment (empyrical) [reply .circleci/config.yml]: Good catch, fixed that up

Comment (TheSavior) [reply .circleci/config.yml]: That only runs checking with the default config which checks .ios.js files but not .android.js files.

**Comment by empyrical** [comment]
I don't know a whole lot about the AppVeyor syntax yet - would just adding more commands after `npm test` work?

**Comment by hramos** [comment]
Landing. The checks can be added to Appveyor's config on a separate PR.

**Pull request closed without merging**